bugsnag-plugin-android-ndk/src/main/jni/utils/stack_unwinder.cpp

```diff
@@ -16,7 +16,7 @@ static unwindstack::Unwinder *crash_time_unwinder;
 // soft lock for using the crash time unwinder - if active, return without
 // attempting to unwind. This isn't a "real" lock to avoid deadlocking in the
 // event of a crash while handling an ANR or the reverse.
-static bool unwinding_crash_stack;
+static volatile bool unwinding_crash_stack;
```

Comment:
Can we use an atomic type here to make absolutely sure? C compilers have a tendency to play fast and loose with volatile.